docbrown/db/src/algorithms/reciprocity.rs

```diff
@@ -0,0 +1,102 @@
+use itertools::Itertools;
+use crate::graph_window::{WindowedGraph, WindowedVertex};
+
+
+fn get_unique_out_neighbours(v: &WindowedVertex) -> Vec<u64> {
+    let mut all_out_neighbours: Vec<u64> = v.out_neighbours_ids().unique().collect();
+    // remove self-loop
+    if let Some(pos) = all_out_neighbours.iter().position(|x| *x == v.g_id) {
```

Comment:
Neighbours probably also shouldn't return yourself, we should fold this into the above function.

Reply:
ok it doesnt